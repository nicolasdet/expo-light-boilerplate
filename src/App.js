import React from 'react';
import { StyleSheet, SafeAreaView } from 'react-native';
import NewsDisplay from './components/NewsDisplay/NewsDisplay';
import HeaderBar from './components/HeaderBar/HeaderBar';
import NewsContextProvider from './store/news-context';
import ModalContextProvider from './store/modal-context';
import AddNewsModal from './components/AddNewsForm/AddNewsModal';

function App() {
  return (
    <SafeAreaView style={styles.container}>
      <NewsContextProvider>
        <ModalContextProvider>
          <HeaderBar />
          <NewsDisplay />
          <AddNewsModal />
        </ModalContextProvider>
      </NewsContextProvider>
    </SafeAreaView>
  );
}

export default App;

const styles = StyleSheet.create({
  header: {
    backgroundColor: 'white',
    alignItems: 'center',
    justifyContent: 'center',
    color: 'black',
  },
  logo: {
    height: '10vh',
    width: '100vw',
  },
  image: {
    height: '40vw',
  },
  title: {
    fontSize: 'calc(10px + 1vh)',
    color: 'black',
    flex: 1,
  },
  container: {
    flex: 1,
    backgroundColor: 'white',
  },
});
