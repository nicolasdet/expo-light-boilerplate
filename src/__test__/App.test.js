import React from 'react';
import { render } from '@testing-library/react-native';
import renderer from 'react-test-renderer';
import App from '../App';

it('renders Home correctly', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Header Text is visible from App', async () => {
  const expectedHeaderText = 'Evenements Culturels Ã  venir';
  const { findByText } = render(<App />);
  const HeaderTextOutput = await findByText(expectedHeaderText);
  expect(HeaderTextOutput).not.toBe(null);
});
