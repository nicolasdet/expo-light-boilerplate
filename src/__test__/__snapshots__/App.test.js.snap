// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Home correctly 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "white",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 90,
          "justifyContent": "center",
          "marginTop": 20,
          "width": "60%",
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
            "marginLeft": 0,
          }
        }
      >
        <Text>
          Evenements Culturels √† venir
        </Text>
      </View>
    </View>
    <View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        hitSlop={
          {
            "bottom": 20,
            "left": 20,
            "right": 20,
            "top": 20,
          }
        }
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "borderColor": "black",
            "borderRadius": 4,
            "borderWidth": 1,
            "justifyContent": "center",
            "padding": 5,
          }
        }
      >
        <Text>
          Ajouter un eÃÅveÃÄnement
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 4,
        "paddingTop": 50,
      }
    }
  >
    <View
      style={
        {
          "flexDirection": "row",
          "justifyContent": "space-around",
          "width": "100%",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "margin": 3,
          }
        }
      >
        <Text>
          Filtrer par dates
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "margin": 3,
          }
        }
      >
        <Text>
          Filtrer par categorie
        </Text>
      </View>
    </View>
    <RCTScrollView
      data={
        [
          {
            "category": "√âv√®nement",
            "date": "2022-06-07",
            "images": [
              "/images/cannes1.jpg",
              "/images/cannes2.jpg",
              "/images/cannes3.jpg",
              "/images/cannes4.jpg",
            ],
            "text": "Les √©quipes du #passCulture sont ravies d'avoir assist√© √† la 1√®re √©dition du festival R√©el les 3,4 & 5 juin √† @villeurbanne, mont√© par des jeunes et pour les jeunes. L'occasion pour 30 envoy√©s sp√©ciaux du #pass de tenir un stand sur le village et profiter des concerts üé∂",
            "title": "Festival R√©el",
          },
          {
            "category": "Utilisateurs",
            "date": "2022-06-06",
            "text": "üé¨ Retour sur le lancement de #lille3000 avec Anne-Rose et Jenny, deux jeunes utilisatrices du #passCulture üöÄL'opportunit√© pour nos b√©n√©ficiaires de faire le plein de d√©couvertes culturelles autour de la th√©matique Utopia !",
            "title": "Utopia",
          },
          {
            "category": "Au bureau",
            "date": "2022-06-03",
            "text": "L'√©quipe du #passCulture est ravie d‚Äôavoir accueilli la ministre de la Culture @RimaAbdulMalak dans ses locaux, ce vendredi 3 juin. Avec l'ensemble des talents ≈ìuvrant au quotidien depuis le lancement du pass, nous avons pu √©changer sur les enjeux √† venir.",
            "title": "Visite de Rima Abdul Malak",
          },
          {
            "category": "Acteurs culturels",
            "date": "2022-06-02",
            "images": [
              "/images/ABF2022.jpg",
            ],
            "text": "#ABF2022 Le pass Culture est pr√©sent aujourd‚Äôhui et demain √† #Metz pour le congr√®s annuel de @abf_info! Un moment pr√©cieux pour √©changer avec les professionnels du secteur et les inviter √† valoriser leurs projets sur l‚Äôapplication üìö",
            "title": "ABF2022",
          },
          {
            "category": "Ambassadeurs",
            "date": "2022-05-31",
            "text": "üé∂ Les ambassadeurs bordelais du #passCulture ont assist√© au concert de @louane et rencontr√© l'artiste √† l'occasion de sa tourn√©e 'Louane en couleurs'.",
            "title": "Louane",
          },
          {
            "category": "Ambassadeurs",
            "date": "2022-05-25",
            "text": "1 an de #passCulture, √ßa repr√©sente quoi pour les jeunes ? Un premier festival, une premi√®re sortie au th√©√¢tre, un tapis rouge, des √©motions sur papier comme sur grand √©cran... Nos ambassadeurs nous d√©voilent les exp√©riences culturelles qu'ils ont pu vivre avec le pass ! ü§©",
            "title": "Anniversaire",
          },
          {
            "category": "√âv√®nement",
            "date": "2022-05-25",
            "images": [
              "/images/cannes1.jpg",
              "/images/cannes2.jpg",
              "/images/cannes3.jpg",
              "/images/cannes4.jpg",
            ],
            "text": "Pour c√©l√©brer le #PassCulture qui souffle sa 1√®re bougie, 14 ambassadeurs lillois du pass Culture ont eu la chance d'√™tre invit√©s au Festival de Cannes et d'√©changer avec la Ministre de la Culture, Rima Abdul Malak avant de monter √† ses c√¥t√©s les marches du Palais des Festivals !",
            "title": "Festival de Cannes",
          },
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={[]}
      viewabilityConfigCallbackPairs={[]}
    >
      <View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              {
                "borderColor": "black",
                "borderWidth": 1,
                "flex": 1,
                "flexDirection": "row",
                "marginHorizontal": 20,
                "marginVertical": 10,
                "padding": 10,
              }
            }
          >
            <View
              style={
                {
                  "flexDirection": "column",
                  "width": "70%",
                }
              }
            >
              <View
                style={
                  {
                    "marginVertical": 2,
                    "padding": 10,
                  }
                }
              >
                <Text>
                  Festival R√©el
                </Text>
              </View>
              <View
                style={
                  {
                    "marginVertical": 2,
                    "padding": 10,
                  }
                }
              >
                <Text>
                  2022-06-07
                </Text>
              </View>
              <View
                style={
                  {
                    "marginVertical": 2,
                    "padding": 10,
                  }
                }
              >
                <Text>
                  Les √©quipes du #passCulture sont ravies d'avoir assist√© √† la 1√®re √©dition du festival R√©el les 3,4 & 5 juin √† @villeurbanne, mont√© par des jeunes et pour les jeunes. L'occasion pour 30 envoy√©s sp√©ciaux du #pass de tenir un stand sur le village et profiter des concerts üé∂
                </Text>
              </View>
            </View>
            <View
              style={
                {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "width": "30%",
                }
              }
            >
              <Image
                source={1}
                style={
                  {
                    "height": 150,
                    "resizeMode": "cover",
                    "width": 150,
                  }
                }
              />
            </View>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              {
                "borderColor": "black",
                "borderWidth": 1,
                "flex": 1,
                "flexDirection": "row",
                "marginHorizontal": 20,
                "marginVertical": 10,
                "padding": 10,
              }
            }
          >
            <View
              style={
                {
                  "flexDirection": "column",
                  "width": "70%",
                }
              }
            >
              <View
                style={
                  {
                    "marginVertical": 2,
                    "padding": 10,
                  }
                }
              >
                <Text>
                  Utopia
                </Text>
              </View>
              <View
                style={
                  {
                    "marginVertical": 2,
                    "padding": 10,
                  }
                }
              >
                <Text>
                  2022-06-06
                </Text>
              </View>
              <View
                style={
                  {
                    "marginVertical": 2,
                    "padding": 10,
                  }
                }
              >
                <Text>
                  üé¨ Retour sur le lancement de #lille3000 avec Anne-Rose et Jenny, deux jeunes utilisatrices du #passCulture üöÄL'opportunit√© pour nos b√©n√©ficiaires de faire le plein de d√©couvertes culturelles autour de la th√©matique Utopia !
                </Text>
              </View>
            </View>
            <View
              style={
                {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "width": "30%",
                }
              }
            />
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              {
                "borderColor": "black",
                "borderWidth": 1,
                "flex": 1,
                "flexDirection": "row",
                "marginHorizontal": 20,
                "marginVertical": 10,
                "padding": 10,
              }
            }
          >
            <View
              style={
                {
                  "flexDirection": "column",
                  "width": "70%",
                }
              }
            >
              <View
                style={
                  {
                    "marginVertical": 2,
                    "padding": 10,
                  }
                }
              >
                <Text>
                  Visite de Rima Abdul Malak
                </Text>
              </View>
              <View
                style={
                  {
                    "marginVertical": 2,
                    "padding": 10,
                  }
                }
              >
                <Text>
                  2022-06-03
                </Text>
              </View>
              <View
                style={
                  {
                    "marginVertical": 2,
                    "padding": 10,
                  }
                }
              >
                <Text>
                  L'√©quipe du #passCulture est ravie d‚Äôavoir accueilli la ministre de la Culture @RimaAbdulMalak dans ses locaux, ce vendredi 3 juin. Avec l'ensemble des talents ≈ìuvrant au quotidien depuis le lancement du pass, nous avons pu √©changer sur les enjeux √† venir.
                </Text>
              </View>
            </View>
            <View
              style={
                {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "width": "30%",
                }
              }
            />
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              {
                "borderColor": "black",
                "borderWidth": 1,
                "flex": 1,
                "flexDirection": "row",
                "marginHorizontal": 20,
                "marginVertical": 10,
                "padding": 10,
              }
            }
          >
            <View
              style={
                {
                  "flexDirection": "column",
                  "width": "70%",
                }
              }
            >
              <View
                style={
                  {
                    "marginVertical": 2,
                    "padding": 10,
                  }
                }
              >
                <Text>
                  ABF2022
                </Text>
              </View>
              <View
                style={
                  {
                    "marginVertical": 2,
                    "padding": 10,
                  }
                }
              >
                <Text>
                  2022-06-02
                </Text>
              </View>
              <View
                style={
                  {
                    "marginVertical": 2,
                    "padding": 10,
                  }
                }
              >
                <Text>
                  #ABF2022 Le pass Culture est pr√©sent aujourd‚Äôhui et demain √† #Metz pour le congr√®s annuel de @abf_info! Un moment pr√©cieux pour √©changer avec les professionnels du secteur et les inviter √† valoriser leurs projets sur l‚Äôapplication üìö
                </Text>
              </View>
            </View>
            <View
              style={
                {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "width": "30%",
                }
              }
            >
              <Image
                source={1}
                style={
                  {
                    "height": 150,
                    "resizeMode": "cover",
                    "width": 150,
                  }
                }
              />
            </View>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              {
                "borderColor": "black",
                "borderWidth": 1,
                "flex": 1,
                "flexDirection": "row",
                "marginHorizontal": 20,
                "marginVertical": 10,
                "padding": 10,
              }
            }
          >
            <View
              style={
                {
                  "flexDirection": "column",
                  "width": "70%",
                }
              }
            >
              <View
                style={
                  {
                    "marginVertical": 2,
                    "padding": 10,
                  }
                }
              >
                <Text>
                  Louane
                </Text>
              </View>
              <View
                style={
                  {
                    "marginVertical": 2,
                    "padding": 10,
                  }
                }
              >
                <Text>
                  2022-05-31
                </Text>
              </View>
              <View
                style={
                  {
                    "marginVertical": 2,
                    "padding": 10,
                  }
                }
              >
                <Text>
                  üé∂ Les ambassadeurs bordelais du #passCulture ont assist√© au concert de @louane et rencontr√© l'artiste √† l'occasion de sa tourn√©e 'Louane en couleurs'.
                </Text>
              </View>
            </View>
            <View
              style={
                {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "width": "30%",
                }
              }
            />
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              {
                "borderColor": "black",
                "borderWidth": 1,
                "flex": 1,
                "flexDirection": "row",
                "marginHorizontal": 20,
                "marginVertical": 10,
                "padding": 10,
              }
            }
          >
            <View
              style={
                {
                  "flexDirection": "column",
                  "width": "70%",
                }
              }
            >
              <View
                style={
                  {
                    "marginVertical": 2,
                    "padding": 10,
                  }
                }
              >
                <Text>
                  Anniversaire
                </Text>
              </View>
              <View
                style={
                  {
                    "marginVertical": 2,
                    "padding": 10,
                  }
                }
              >
                <Text>
                  2022-05-25
                </Text>
              </View>
              <View
                style={
                  {
                    "marginVertical": 2,
                    "padding": 10,
                  }
                }
              >
                <Text>
                  1 an de #passCulture, √ßa repr√©sente quoi pour les jeunes ? Un premier festival, une premi√®re sortie au th√©√¢tre, un tapis rouge, des √©motions sur papier comme sur grand √©cran... Nos ambassadeurs nous d√©voilent les exp√©riences culturelles qu'ils ont pu vivre avec le pass ! ü§©
                </Text>
              </View>
            </View>
            <View
              style={
                {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "width": "30%",
                }
              }
            />
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              {
                "borderColor": "black",
                "borderWidth": 1,
                "flex": 1,
                "flexDirection": "row",
                "marginHorizontal": 20,
                "marginVertical": 10,
                "padding": 10,
              }
            }
          >
            <View
              style={
                {
                  "flexDirection": "column",
                  "width": "70%",
                }
              }
            >
              <View
                style={
                  {
                    "marginVertical": 2,
                    "padding": 10,
                  }
                }
              >
                <Text>
                  Festival de Cannes
                </Text>
              </View>
              <View
                style={
                  {
                    "marginVertical": 2,
                    "padding": 10,
                  }
                }
              >
                <Text>
                  2022-05-25
                </Text>
              </View>
              <View
                style={
                  {
                    "marginVertical": 2,
                    "padding": 10,
                  }
                }
              >
                <Text>
                  Pour c√©l√©brer le #PassCulture qui souffle sa 1√®re bougie, 14 ambassadeurs lillois du pass Culture ont eu la chance d'√™tre invit√©s au Festival de Cannes et d'√©changer avec la Ministre de la Culture, Rima Abdul Malak avant de monter √† ses c√¥t√©s les marches du Palais des Festivals !
                </Text>
              </View>
            </View>
            <View
              style={
                {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "width": "30%",
                }
              }
            >
              <Image
                source={1}
                style={
                  {
                    "height": 150,
                    "resizeMode": "cover",
                    "width": 150,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    presentationStyle="overFullScreen"
    style={
      {
        "flex": 1,
      }
    }
    transparent={true}
    visible={false}
  />
</RCTSafeAreaView>
`;
